.github/workflows/idempotency-tests.yml

```diff
@@ -0,0 +1,114 @@
+name: idempotency-tests
```

Comment:
I'm curious to why you decided to add a new workflow definition as opposed to keeping that stage on the integration tests one? we're paying the cost of building kiota twice.

Reply:
That's correct, but I find that the current results + uploaded artifacts page is too bloated ( [e.g.](https://github.com/microsoft/kiota/actions/runs/6272467707) ).
Separating the workflows it's easier to navigate and read both results and find the relevant artifacts.

I understand this is subjective, I can revert if desired, but I find the new report to provide a better UX for us developers: [e.g.](https://github.com/microsoft/kiota/actions/runs/6266455834)

Reply:
that makes perfect sense. I too have struggled with parsing the logs of the integration tests on the main summary view. Resolving.